Replace Spark/DuckDB APIs with Ibis (#414)
dsgrid/dsgrid: docs/source/user_guide/project_queries/query_concepts.md

   - Pre-filter dataset according to the `dimension_filters` in the dataset data model of the query.
   - Map dataset dimensions to the project.
   - Convert units.

>    - Evaluate the Ibis query by writing the table to the filesystem in the directory `query_output/cached_project_mapped_datasets`.

Reviewer: ```suggestion
   - Persist the mapped dataset by writing the table to the filesystem in the directory `query_output/cached_project_mapped_datasets`.
```

     - dsgrid restarts the runtime session for this query. If custom runtime configuration parameters are defined in the `runtime_conf_per_dataset` data model of the query for this dataset, dsgrid will apply them in the new session.
     - If the backend is Spark, the value `spark.sql.shuffle.partitions` may need to be increased for very large datasets.

:::{note}

> Currently, there is no way to skip caching of the dataset. If it is not performed, the backend query can grow too large and take too long to complete in normal compute node allocations.

Reviewer: ```suggestion
Currently, there is no way to skip mapped dataset caching. If it is not performed, the backend query can grow too large and take too long to complete in normal compute node allocations.
```

:::

3. **Combine datasets** - Combine the datasets as specified by the `expression` in the dataset data model of the query. The default is to take a union of all datasets.

> 4. **Persist intermediate table** - If the option `--persist-intermediate-table` is `true` (which is the default) then dsgrid will evaluate the Ibis query from the previous step by writing the table to the filesystem in the directory `query_output/cached_tables`. This can be disabled by setting `--no-persist-intermediate-table`.

Reviewer: ```suggestion
4. **Persist intermediate table** - If the option `--persist-intermediate-table` is `true` (which is the default) then dsgrid will evaluate the query from the previous step by writing the table to the filesystem in the directory `query_output/cached_tables`. This can be disabled by setting `--no-persist-intermediate-table`.
```

Reviewer: https://github.com/dsgrid/dsgrid/pull/414/commits/b36fab2cffcf3d0327d282902fd95f83905412f2

5. **Apply result filters** - Apply any dimension_filters defined in the `result` data model of the query.

6. **Apply aggregations/disaggregations** - Apply any aggregations or disaggregations defined in the `result` data model of the query.

7. **Replace IDs with names** - If the field `replace_ids_with_names` in the `result` data model is `true`, replace all dimension record IDs with names.

8. **Sort columns** - If the field `sort_columns` in the `result` data model is `true`, sort the table by those columns.

> 9. **Write output** - Evaluate the Ibis query for the previous steps by writing the table to the filesystem in the directory `query_output/<query-name>`.

Reviewer: Per suggestion above, can these `Spark job` to `Ibis query` replacements all be changed to something like `query`? (And are these descriptions only applicable with Spark, or do they also apply to DuckDB?)
